goal_src/engine/debug/menu.gc

```diff
@@ -5,3 +5,1968 @@
+   v0-0
+   )
+  )
+
+;; definition of type debug-menu-item-var
+(deftype debug-menu-item-var (debug-menu-item)
+  ((display-str      string          :offset-assert  24)
+   (grabbed-joypad-p basic           :offset-assert  28)
+   (float-p          basic           :offset-assert  32)
+   (range-p          basic           :offset-assert  36)
+   (show-len         int32           :offset-assert  40)
+   (inc-delay        int32           :offset-assert  44)
+   (inc-delay-ctr    int32           :offset-assert  48)
+   (step-delay-ctr   int32           :offset-assert  52)
+   (inc-dir          int32           :offset-assert  56)
+   (fval             float           :offset-assert  60)
+   (fundo-val        float           :offset-assert  64)
+   (frange-min       float           :offset-assert  68)
+   (frange-max       float           :offset-assert  72)
+   (fstart-inc       float           :offset-assert  76)
+   (fstep            float           :offset-assert  80)
+   (fprecision       int32           :offset-assert  84)
+   (factivate-func   function        :offset-assert  88)
+   (ival             int32           :offset         60)
+   (iundo-val        int32           :offset         64)
+   (irange-min       int32           :offset         68)
+   (irange-max       int32           :offset         72)
+   (istart-inc       int32           :offset         76)
+   (istep            int32           :offset         80)
+   (ihex-p           basic           :offset-assert  92)
+   (iactivate-func   (function int)  :offset         88)
+   (ifloat-p         basic           :offset-assert  96)
+   )
+  :method-count-assert 9
+  :size-assert         #x64
+  :flag-assert         #x900000064
+  (:methods
+    (new (symbol type string int int) _type_ 0)
+    )
+  )
+
+(define-extern debug-menu-context-send-msg (function debug-menu-context int int debug-menu-context))
+(define-extern debug-menu-make-from-template (function debug-menu-context pair debug-menu-node))
+(define-extern debug-menu-item-send-msg (function debug-menu-item int debug-menu-item))
+(define-extern debug-menu-item-var-render (function debug-menu-item-var int int int symbol debug-menu-item-var))
+(define-extern debug-menu-send-msg (function debug-menu int symbol debug-menu))
+
+;; definition for function debug-menu-item-var-update-display-str
+(defun debug-menu-item-var-update-display-str ((arg0 debug-menu-item-var))
+  (cond
+   ((-> arg0 float-p)
+    (format (clear (-> arg0 display-str)) "~f" (-> arg0 fval))
+    )
+   ((-> arg0 ihex-p)
+    (format (clear (-> arg0 display-str)) "x~X" (-> arg0 fval))
+    )
+   ((-> arg0 ifloat-p)
+    (cond
+     ((and
+       (< (the-as int (-> arg0 fval)) 0)
+       (< -100 (the-as int (-> arg0 fval)))
+       )
+      (let ((s5-2 format)
+            (a0-8 (clear (-> arg0 display-str)))
+            (a1-2 "-0.~1d")
+            (v1-8 (abs (the-as int (-> arg0 fval))))
+            )
+       (s5-2 a0-8 a1-2 (/ (mod v1-8 100) 10))
```

Comment:
hmm, this looks weird.  I'd like to take a look at how the decompiler handles `mod` here first.